### PR TITLE
🌱 Prepare for v2 beta1 release

### PR DESCRIPTION
**Pull request by azeemshaikh38**
Commit created from this PR will be used to generate the pre-release of scorecard-action. I've decided to name the tag - `v2.0.0-beta.1`.

### CODE REVIEW

File: action.yaml [hunk 1]
```diff
@@ -48,6 +48,6 @@ branding:
 
 runs:
   using: "docker"
-  image: "docker://gcr.io/openssf/scorecard-action:latest"
+  image: "docker://gcr.io/openssf/scorecard-action:v2.0.0-beta1"
```
Comment (justaugustus) [inline action.yaml:51]:
```suggestion
  image: "docker://gcr.io/openssf/scorecard-action:v2.0.0-beta.1"
```

Needs a `.` after `beta` to be a SemVer-compliant tag.

**Review by justaugustus — CHANGES_REQUESTED** [review]

**Comment by justaugustus** [comment]
(Let's also see if we can drain the remaining dependabot updates before cutting the release)

**Comment by azeemshaikh38** [comment]
> (Let's also see if we can drain the remaining dependabot updates before cutting the release)

Fair point. Looking into it.

File: RELEASE.md [hunk 2]
```diff
@@ -30,6 +30,32 @@ We're not striving for perfection with the template, but the tracking issue
...
+### Update the scorecard-action version
+
+NOTE: we have a chicken-and-egg problem where the commit to be used for the release
+needs to have the image tag that only gets created *after* the commit is pushed. We
+workaround that by pre-selecting and referencing the image tag instead of the SHA which isn't ideal 
+but workable.
+
+Pre-select the tag which will be used for the release. For this document, we'll use: `Tag`.
+
+Update the image tage in [action.yaml](action.yaml) to use `Tag`.
```
Comment (laurentsimon) [inline RELEASE.md:47]: nit: tage

File: action.yaml [hunk 3]
```diff
@@ -48,6 +48,6 @@ branding:
 
 runs:
   using: "docker"
-  image: "docker://gcr.io/openssf/scorecard-action:latest"
+  image: "docker://gcr.io/openssf/scorecard-action:v2.0.0-beta.1"
```
Comment (laurentsimon) [inline action.yaml:51]: this will be replaced by a hash, like `:tag@sha256:xxx` after?

**Review by laurentsimon — APPROVED** [review]

**Comment by naveensrinivasan** [comment]
> > > (Let's also see if we can drain the remaining dependabot updates before cutting the release)
> > 
> > 
> > Fair point. Looking into it.
> 
> Getting errors running `go mod tidy`:
> 
> ```
> go mod tidy
> go: finding module for package go.opentelemetry.io/otel/metric/registry
> go: finding module for package go.opentelemetry.io/otel/semconv
> github.com/ossf/scorecard-action/signing imports
>         github.com/sigstore/cosign/cmd/cosign/cli/sign imports
>         github.com/sigstore/cosign/cmd/cosign/cli/fulcio/fulcioverifier/ctl imports
>         github.com/google/certificate-transparency-go imports
>         go.etcd.io/etcd/v3 imports
>         go.etcd.io/etcd/tests/v3/integration imports
>         go.etcd.io/etcd/server/v3/embed imports
>         go.opentelemetry.io/otel/semconv: module go.opentelemetry.io/otel@latest found (v1.8.0), but does not contain package go.opentelemetry.io/otel/semconv
> github.com/ossf/scorecard-action/signing imports
>         github.com/sigstore/cosign/cmd/cosign/cli/sign imports
>         github.com/sigstore/cosign/cmd/cosign/cli/fulcio/fulcioverifier/ctl imports
>         github.com/google/certificate-transparency-go imports
>         go.etcd.io/etcd/v3 imports
>         go.etcd.io/etcd/tests/v3/integration imports
>         go.etcd.io/etcd/server/v3/embed imports
>         go.opentelemetry.io/otel/exporters/otlp imports
>         go.opentelemetry.io/otel/sdk/metric/controller/basic imports
>         go.opentelemetry.io/otel/metric/registry: module go.opentelemetry.io/otel/metric@latest found (v0.31.0), but does not contain package go.opentelemetry.io/otel/metric/registry
> ```
> 
> Anyone have an idea?

Is this on your branch?

**Comment by azeemshaikh38** [comment]
> Is this on your branch?

No, this is on HEAD.

File: RELEASE.md (hunk 2 above)
Comment (azeemshaikh38) [reply RELEASE.md]: Done, thanks.

**Comment by naveensrinivasan** [comment]
> > Is this on your branch?
> 
> No, this is on HEAD.

Looks like this is the problem https://github.com/open-telemetry/opentelemetry-go/issues/2938#issuecomment-1156255999

**Comment by azeemshaikh38** [comment]
> > > Is this on your branch?
> > 
> > 
> > No, this is on HEAD.
> 
> Looks like this is the problem [open-telemetry/opentelemetry-go#2938 (comment)](https://github.com/open-telemetry/opentelemetry-go/issues/2938#issuecomment-1156255999)

Thanks, that's a great find! I'll try to see if I can use the comments there to resolve our dependency problem.

**Comment by naveensrinivasan** [comment]
> > > > Is this on your branch?
> > > 
> > > 
> > > No, this is on HEAD.
> > 
> > 
> > Looks like this is the problem [open-telemetry/opentelemetry-go#2938 (comment)](https://github.com/open-telemetry/opentelemetry-go/issues/2938#issuecomment-1156255999)
> 
> Thanks, that's a great find! I'll try to see if I can use the comments there to resolve our dependency problem.

It is a `replace` directive as the project has broken the API, and the downstream hasn't changed. This is dependency hell.

**Comment by naveensrinivasan** [comment]
> > (Let's also see if we can drain the remaining dependabot updates before cutting the release)
> 
> Fair point. Looking into it.

Most of them have been merged!

**Comment by justaugustus** [comment]
> @justaugustus blocked on your approval. Dependabot alerts are now fixed.

Azeem, can you investigate the unit test failures?
I believe it's related to the signing logic.

**Comment by azeemshaikh38** [comment]
> > @justaugustus blocked on your approval. Dependabot alerts are now fixed.
> 
> Azeem, can you investigate the unit test failures? I believe it's related to the signing logic.

They are intermittent failures since the issue is with the REST API the test talks to (its a mock request, so the failures are only happening on the test but prod works fine). The fix is going to be on the API side, no changes needed here. Will retry the test to get it to pass.

**Comment by azeemshaikh38** [comment]
> > @justaugustus blocked on your approval. Dependabot alerts are now fixed.
> 
> Azeem, can you investigate the unit test failures? I believe it's related to the signing logic.

Fixed now with the retry :)

**Comment by azeemshaikh38** [comment]
> > > @justaugustus blocked on your approval. Dependabot alerts are now fixed.
> > 
> > 
> > Azeem, can you investigate the unit test failures? I believe it's related to the signing logic.
> 
> Fixed now with the retry :)

@justaugustus friendly ping.

**Review by justaugustus — APPROVED** [review]

**Comment by justaugustus** [comment]
> @justaugustus friendly ping.

Thanks for the details on the failure.
Approved!

**Pull request merged**